gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/authentication/provider/security/ManagementAuthenticationProvider.java

```diff
@@ -148,10 +162,18 @@ public Authentication authenticate(Authentication authentication) throws Authent
             authenticationToken.setDetails(details);
             return authenticationToken;
         }
-
+        if (userException == userNotFoundException) {
+            //Didn't find user in any IDP (Mongo or JDBC), so no password encoding has been proceeded.
+            doFakePasswordEncoding();
+        }
         throw new BadCredentialsException("No user found for providers " + StringUtils.collectionToDelimitedString(identities, ","));
     }
 
+    private void doFakePasswordEncoding(){
+        //PEN-21 Encoding password takes a while. To ensure execution time the same for not existing user, introduced fake password checking.
+        bCryptPasswordEncoder.matches(UUID.randomUUID().toString(), "$2a$10$hdjt9YGrSudbIljTqAtcW.KOxNJscq00Nxv088wPy6GDKXCJe0aCm");
```

Comment:
question: we discussed it in meeting so I understand why the UUID is used but should we align the behaviour with the GW which is evaluating the password provided by the user. Maybe it is more realistic to test the pwd provided by the user here too ? :thinking: